### ISSUE TITLE
New rule: unused generic type parameter

### ISSUE THREAD

**Issue by lostfictions** [palantir/tslint#1126]
For example, in the [Redux type definitions](https://github.com/reactjs/redux/blob/324849df224acdfdf88a17bce97ec400031a705c/index.d.ts#L96-L98) we see this:

``` typescript
export interface Dispatch<S> {
    <A extends Action>(action: A): A;
}
```

`S` is unused and only adds confusion and friction for consumers of this interface! It'd be useful to be able to show a warning in cases like this.

**Comment by myitcv** [comment]
👍 

For some reason I thought there was already an open issue for this, but I can't find one so thanks for opening @lostfictions 

For useful background on the issues this creates see https://github.com/Microsoft/TypeScript/issues/7797 amongst other issues

**Comment by andy-hanson** [comment]
Should be fixed by #2235, as the TypeScript implementation of no-unused-locals is used now.
